### DISCUSSION TITLE
Using CCF for CO2 Emission by a specific Process

### DISCUSSION THREAD

**Discussion by tushartale** [cloud-carbon-footprint/cloud-carbon-footprint#1248] [Q&A] [answered]
Hi,

I have been playing around with CCF for on-premise Carbon emission calculations.  I could calculate carbon emission at machine level but requirement is more granular  - want to calculate Carbon emission by a specific process (Hosted on Window OS)

By using same logic for machine level calculation ; I felt that min-watt value if ignored ;same formula can be used for calculating carbon emission for a process if we know average CPU utilization for a process along with duration process was running. 

So PROCESS CO2 emission formula will look like below for sample process name XYZ
Average Watts =  Avg vCPU Utilization for a process XYZ  * (Max Watts - Min Watts)
Process Watt-Hours = Average Watts * vCPU Hours (Time for which XYZ Process was running)

Is this correct way to use this formula for a process or I am doing something wrong here ?

Please suggest.

Thanks

**Comment by camcash17** [comment, marked as answer]
Hi @tushartale, it looks like you are on the right track and you are correct, currently CCF on-premise estimations support Machine type (Ie. server, laptop, desktop). I believe it could be possible to differentiate by a specific process rather than machine type. The schema offers an optional param for `machineName` which can be used to filter by OS.

If you might be interested in creating a PR to update the logic to handle inputting custom vCPU utilizations for different processes, we'd be happy to help support.
